### PR TITLE
Fix `IField.required` to not be required by default.

### LINKED ISSUE #104: IField.required attribute is True but must not.
### What I did:

We use `IField` https://github.com/zopefoundation/zope.schema/blob/master/src/zope/schema/interfaces.py#L192 with z3c.form to generate a form (plone.schemaeditor). 

### What I expect to happen:

The `required` of above schema interface attribute should render a field I can save checked or unchecked  https://github.com/zopefoundation/zope.schema/blob/master/src/zope/schema/interfaces.py#L192

### What actually happened:

The field can only be saved as selected, because it is required.

### What version of Python and Zope/Addons I am using:

Plone with recent zc3.form

## Analysis:

`IField.required` is a Bool field with `required=True`. This z3c.form does what it is told and render the field required.

A Bool fields default for required is `True`. It is inherited from `Field` https://github.com/zopefoundation/zope.schema/blob/master/src/zope/schema/_bootstrapfields.py#L222

## Solution

Explicitly set `IField.required` `Bool` field definition to `False` like so:

```
    required = Bool(
        title=_("Required"),
        description=(_("Tells whether a field requires its value to exist.")),
        required=False,
        default=True)
```

### PR DESCRIPTION
**Pull request by agitator**
Fixes #104

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -5,8 +5,8 @@
 6.0.2 (unreleased)
 ==================
 
-- Nothing changed yet.
-
+- Fix IField Bool to not be required by default.
```
Comment (icemac) [inline CHANGES.rst:8]:
```suggestion
- Fix ``IField.required`` to not be required by default. 
```

**Review by icemac — APPROVED** [review]
LGTM.

**Review by jamadden — APPROVED** [review]
This seems like potentially a large semantic change (not everything pays attention to defaults), suggesting to me that the next release should be 6.1.0.

Could there be a test added to check `not IField['required'].required`? Otherwise this could easily be changed back.

**Comment by agitator** [comment]
double checked in Plone content type schema editor - works now as expected

**Review by jensens — APPROVED** [review]
LGTM 

Anyway, self referencing implementations are hard. In interfaces.py line 410 `required=False` is redundant, but anyway, for the sake of being explicit (and documentation) I think this is fine.

**Comment by pbauer** [comment]
This breaks the schema in eea.facettednavigation in a way that I do not understand:

```
2021-02-03 13:34:23,251 INFO    [chameleon.config:38][MainThread] directory cache: /Users/pbauer/workspace/dynajet_relaunch/var/cache.
Traceback (most recent call last):
  File "/Users/pbauer/.cache/buildout/eggs/zope.configuration-4.4.0-py3.8.egg/zope/configuration/config.py", line 1687, in toargs
    args[str(name)] = field.fromUnicode(s)
  File "/Users/pbauer/.cache/buildout/eggs/zope.configuration-4.4.0-py3.8.egg/zope/configuration/fields.py", line 172, in fromUnicode
    value = self.context.resolve(name)
  File "/Users/pbauer/.cache/buildout/eggs/zope.configuration-4.4.0-py3.8.egg/zope/configuration/config.py", line 226, in resolve
    __import__(mname)
  File "/Users/pbauer/workspace/dynajet_relaunch/src-mrd/eea.facetednavigation/eea/facetednavigation/subtypes/interfaces.py", line 25, in <module>
    class IFacetedSubtyper(Interface):
  File "/Users/pbauer/workspace/dynajet_relaunch/src-mrd/eea.facetednavigation/eea/facetednavigation/subtypes/interfaces.py", line 29, in IFacetedSubtyper
    can_enable = schema.Bool(u'Can enable faceted navigation',
  File "/Users/pbauer/workspace/dynajet_relaunch/src-mrd/zope.schema/src/zope/schema/_bootstrapfields.py", line 614, in __init__
    super(Bool, self).__init__(required=required, **kw)
  File "/Users/pbauer/workspace/dynajet_relaunch/src-mrd/zope.schema/src/zope/schema/_bootstrapfields.py", line 269, in __init__
    self.required = required
  File "/Users/pbauer/workspace/dynajet_relaunch/src-mrd/zope.schema/src/zope/schema/fieldproperty.py", line 84, in __set__
    field.validate(value)
  File "/Users/pbauer/workspace/dynajet_relaunch/src-mrd/zope.schema/src/zope/schema/_bootstrapfields.py", line 300, in validate
    self._validate(value)
  File "/Users/pbauer/workspace/dynajet_relaunch/src-mrd/zope.schema/src/zope/schema/_bootstrapfields.py", line 621, in _validate
    Field._validate(self, value)
  File "/Users/pbauer/workspace/dynajet_relaunch/src-mrd/zope.schema/src/zope/schema/_bootstrapfields.py", line 349, in _validate
    raise WrongType(
zope.schema._bootstrapinterfaces.WrongType: ('Can enable faceted navigation', <class 'bool'>, 'required')
```

**Comment by pbauer** [comment]
@agitator: Thanks for fixing my problem.

**Comment by icemac** [comment]
Is there still something to be done before this PR can be merged?

**Comment by agitator** [comment]
no other problems found so far, should be fine

**Comment by icemac** [comment]
Just released as https://pypi.org/project/zope.schema/6.1.0/ to PyPI.

**Pull request merged**